### PR TITLE
Add token-file option to 'package-registry login'

### PR DESCRIPTION
**Pull request by yim-lee**
rdar://109372471

### CODE REVIEW

**Comment by yim-lee** [comment]
cc @Palleas

**Comment by yim-lee** [comment]
@swift-ci please smoke test

File: Sources/PackageRegistryTool/PackageRegistryTool+Auth.swift [hunk 1]
```diff
@@ -145,6 +151,10 @@ extension SwiftPackageRegistryTool {
                 if let token {
                     // User provided token
                     storePassword = token
+                } else if let tokenFilePath {
+                    print("Reading access token from \(tokenFilePath).")
+                    storePassword = try localFileSystem.readFileContents(tokenFilePath).cString
```
Comment (tomerd) [inline Sources/PackageRegistryTool/PackageRegistryTool+Auth.swift:156]: readFileContents can return String, no need to use `.cString`

**Comment by yim-lee** [comment]
@swift-ci please smoke test

**Comment by yim-lee** [comment]
@swift-ci please test Windows platform

**Review by neonichu — APPROVED** [review]

**Pull request merged**